ios/CameraView+TakePhoto.swift

```diff
@@ -54,29 +54,64 @@ extension CameraView {
       }
 
       // Create photo settings
-      let photoSettings = AVCapturePhotoSettings(format: format)
+      var photoSettings: AVCapturePhotoSettings?
+
+      if #available(iOS 14.3, *), self.enableRawCapture?.boolValue == true {
+        let query =
+          photoOutput.isAppleProRAWEnabled ? { AVCapturePhotoOutput.isAppleProRAWPixelFormat($0) } : { AVCapturePhotoOutput.isBayerRAWPixelFormat($0) }
+
+        // Retrieve the RAW format, favoring Apple ProRAW when it's in an enabled state.
+        guard let rawFormat =
+          photoOutput.availableRawPhotoPixelFormatTypes.first(where: query) else {
+          fatalError("No RAW format found.")
+        }
+
+        let processedFormat = [AVVideoCodecKey: AVVideoCodecType.hevc]
+
+        photoSettings = AVCapturePhotoSettings(rawPixelFormatType: rawFormat,
+                                               processedFormat: processedFormat)
+      } else {
+        photoSettings = AVCapturePhotoSettings(format: format)
+      }
 
       // default, overridable settings if high quality capture was enabled
       if self.enableHighQualityPhotos?.boolValue == true {
-        photoSettings.isHighResolutionPhotoEnabled = true
+        photoSettings!.isHighResolutionPhotoEnabled = true
         if #available(iOS 13.0, *) {
-          photoSettings.photoQualityPrioritization = .quality
+          photoSettings!.photoQualityPrioritization = .quality
         }
       }
 
+      if #available(iOS 12.0, *), self.enableRawCapture?.boolValue == true {
+        // Select a supported thumbnail codec type and thumbnail dimensions
+        guard let thumbnailPhotoCodecType = photoSettings!.availableRawEmbeddedThumbnailPhotoCodecTypes.first else {
+          // Handle failure to find an available thumbnail photo codec type
+          ReactLogger.log(level: .info, message: "Error finding an available thumbnail photo codec type")
+          return
+        }
+
+        let dimensions = videoDeviceInput.device.activeFormat.highResolutionStillImageDimensions
+
+        photoSettings!.rawEmbeddedThumbnailPhotoFormat = [
+          AVVideoCodecKey: thumbnailPhotoCodecType,
+          AVVideoWidthKey: dimensions.width,
+          AVVideoHeightKey: dimensions.height,
+        ]
```

Comment:
Same as above - this could probably go into a `AVCapturePhotoSettings` wrapper class instead of here, no?